### ISSUE TITLE
MissingPluginException(No implementation found for method getTemporaryDirectory on channel plugins.flutter.io/path_provider)

### ISSUE THREAD

**Issue by nicoroy2561** [flutter/flutter#84598]
## Steps to Reproduce
1. Create any app that uses a plugin using path_provider's getTemporaryDirectory (or just call it directly in an empty project's main.dart)
2. Run the app in an android 5/5.1 emulator -> it will not work due to MissingPluginException
3. Run the app in an android 7+ emulator (6 untested as of now) -> it will work just fine

**Expected results:**
The program running just fine, as it does on any newer Android version, wether it is emulated or not.

**Actual results:**
```
Installing build/app/outputs/flutter-apk/app.apk...                13.8s
D/EGL_emulation( 5766): eglMakeCurrent: 0x7fbd44cacd80: ver 2 0 (tinfo 0x7fbd47c88d40)
E/flutter ( 5766): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: MissingPluginException(No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider)
E/flutter ( 5766): #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:156:7)
E/flutter ( 5766): <asynchronous suspension>                            
E/flutter ( 5766): #1      getApplicationDocumentsDirectory (package:path_provider/path_provider.dart:138:24)
E/flutter ( 5766): <asynchronous suspension>                            
E/flutter ( 5766): #2      HiveX.initFlutter (package:hive_flutter/src/hive_extensions.dart:11:20)
E/flutter ( 5766): <asynchronous suspension> 
```

<details>

```
Analyzing test...    
No issues found! (ran in 7.3s)
```

```
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.2.3 20D91 darwin-x64, locale en-GB)
    • Flutter version 2.2.2 at /Users/__/flutter
    • Framework revision d79295af24 (3 days ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/__/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/__/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.57.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (2 available)
    • Android SDK built for x86 64 (mobile) • emulator-5554 • android-x64    • Android 5.1.1 (API 22) (emulator)
    • Chrome (web)                          • chrome        • web-javascript • Google Chrome 91.0.4472.101

• No issues found!
```

</details>

Notice that this is already happening in our deployed app: the update we released some 4 days ago is not working on Android 5 devices.
None of the build options (--debug, --profile, --release) makes any difference.

**What I have tried:**
1. Running after flutter clean / flutter pub upgrade
2. [This](https://github.com/flutter/flutter/issues/78625#issuecomment-803243381)

None of the above worked.

**Comment by nicoroy2561** [comment]
<

**Comment by nicoroy2561** [comment]
d

**Comment by nicoroy2561** [comment]
Sorry for that. My keyboard was acting strange.

**Comment by nicoroy2561** [comment]
After one day of looking at files and comparing them, turns out it was one of the plugins (audio_session) that, on version 0.1.4, apparently broke something (later fixed in v0.1.5).
While it was a matter of simply running "flutter pub upgrade" to fix it (because it was luckily patched soon after, apparently), it feels weird to me that there was no sign anywhere pointing to where the issue was.
Either way, I managed to fix this so I'll close the issue.

**Comment by nicoroy2561** [comment]
@TahaTesser as requested in the other [issue](https://github.com/flutter/flutter/issues/83010#issuecomment-872101462), here's the tag. I opened this shortly after and went on finding the issue and closing it myself, so all good.
Thank you for your time!
